cockatrice/src/userinfobox.cpp

```diff
@@ -215,9 +232,42 @@ void UserInfoBox::actPassword()
     if (!dlg.exec())
         return;
 
+    auto oldPassword = dlg.getOldPassword();
+    auto newPassword = dlg.getNewPassword();
+
+    if (client->getServerSupportsPasswordHash()) {
+        Command_RequestPasswordSalt cmd;
+        cmd.set_user_name(client->getUserName().toStdString());
+
+        PendingCommand *pend = client->prepareSessionCommand(cmd);
+        connect(pend,
```

Comment:
What is this doing?

Reply:
it asks the salt before using changePassword